crates/bevy_render/src/render_resource/buffer_vec.rs

```diff
@@ -511,3 +511,181 @@ where
         }
     }
 }
+
+/// A structure for storing raw bytes that have already been properly formatted
+/// for use by the GPU.
+///
+/// "Properly formatted" means that item data already meets the alignment and padding
+/// requirements for how it will be used on the GPU. The item type must implement [`NoUninit`]
+/// for its data representation to be directly copyable.
+///
+/// The contained data is stored in system RAM. Calling [`reserve`](RawBufferVec::reserve)
+/// allocates VRAM from the [`RenderDevice`].
+/// [`write_buffer`](RawBufferVec::write_buffer) queues copying of the data
+/// from system RAM to VRAM.
+///
+/// Unlike [`RawBufferVec`] this will add padding to respect alignment rules if necessary when
+/// adding data to it. This is useful when working with buffers that have alignment rules
+/// like uniform buffers.
+pub struct AlignedRawBufferVec<T: NoUninit> {
```

Comment:
Maybe call it DynamicRawBufferVec to follow our conventions? Although I never really liked them, DynamicOffsetRawBufferVec is a bit long. :)

Reply:
I don't feel like it would represent what it is correctly. It's very much a `RawBufferVec` but with padding for offset alignment rules. Dynamic doesn't really imply that to me. I never liked the `Dynamic*` types because it never made sense to me why they would be more dynamic than the non dynamic types.

Also, with the eventual goal to remove/replace encase, these Dynamic* types would most likely disappear since they only exist to wrap encase stuff.